File: 0118-pascals-triangle/0118-pascals-triangle.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:

        numRows -= 1
        if numRows == 0: return [[1]]
        if numRows == 1: return [[1], [1,1]]

        res = [[1], [1,1]]

        for i in range(2, numRows + 1):
            temp = [1]
            for j in range(i - 1):
                temp.append(res[i - 1][j] + res[i - 1][j+1])
            temp.append(1)
            res.append(temp)
        

        return res

        

File: 0118-pascals-triangle/0118-pascals-triangle.cs
public class Solution {
    public IList<IList<int>> Generate(int numRows) {
        var res = new List<IList<int>>();
        for (int i = 1; i <= numRows; i++) {
            var temp = new List<int>();
            for (int j = 1; j <= i; j++) {
                
                if (j == 1 || j == i) temp.Add(1);
                else {
                    temp.Add(res[i - 2][j - 2] + res[i - 2][j-1]);
                } 
                
            }
            res.Add(temp);
        }
        return res;
    }
}